src/test/java/cs321/btree/BTreeTest.java

```diff
@@ -454,6 +480,62 @@ private boolean validateSearchTree(BTree b) {
     }
 
 
+    /**
+     * Checks BTree b against an array of keys to ensure that they were inserted into the BTree.
+     * The important check here is that no keys are being lost during complex
+     * operations such as splitting children in the tree.
+     *
+     * Also used to validate that all the keys in the BTree
+     * are sorted by using an in order traversal of the tree
+     *
+     * IMPORTANT: handles duplicates in inputKeys by removing them.
+     *
+     * @return true if BTree in order traversal matches provide input
+     */
+    private boolean validateBTreeInserts(BTree b, long[] inputKeys) {
```

Comment:
Uglier than intended due to input potentially containing duplicates. Will consider refactoring with more time.